crates/executor/src/executor.rs

```diff
@@ -409,24 +404,43 @@ pub fn execute<DB: StateProvider>(
         n if n >= config.spec_upgrades.byzantium => Some(WEI_3ETH),
         _ => Some(WEI_5ETH),
     }
-    .map(|reward| {
-        // add block reward to beneficiary/miner
-        if let Some(beneficiary) = beneficiary {
-            // if account is present append `Changed` changeset for block reward
-            let old = to_reth_acc(&beneficiary);
-            let mut new = old;
-            new.balance += U256::from(reward);
-            BTreeMap::from([(header.beneficiary, AccountInfoChangeSet::Changed { new, old })])
-        } else {
-            // if account is not present append `Created` changeset
-            BTreeMap::from([(
-                header.beneficiary,
-                AccountInfoChangeSet::Created {
-                    new: Account { nonce: 0, balance: reward.into(), bytecode_hash: None },
-                },
-            )])
+    .map(|reward| -> Result<_, _> {
+        let mut reward_beneficiaries: BTreeMap<H160, u128> = BTreeMap::new();
+        // Calculate Uncle reward
+        for ommer in ommers {
+            let ommer_reward = ((8 + ommer.number - header.number) as u128 * reward) >> 3;
+            *reward_beneficiaries.entry(ommer.beneficiary).or_default() += ommer_reward;
         }
-    });
+        // insert main block reward
+        *reward_beneficiaries.entry(header.beneficiary).or_default() +=
+            reward + (reward >> 5) * ommers.len() as u128;
```

Comment:
can you add a link to the spec pls

Reply:
Added the passage from yellowpaper, and OpenEthereum link to code that does this.